lib/main.py

```diff
@@ -27,19 +29,27 @@ async def lifespan(_app: FastAPI):
 
 
 APP = FastAPI(lifespan=lifespan)
-TASK_LIST: queue.Queue = queue.Queue(maxsize=100)
 
 
 class BackgroundProcessTask(threading.Thread):
     def run(self, *args, **kwargs):  # pylint: disable=unused-argument
+        nc = NextcloudApp()
+        # Hack to workaround auth problems
+        nc.set_user("admin")
+
         while True:
-            task = TASK_LIST.get(block=True)
+            response = nc.providers.task_processing.next_task("core:text2text")
+            if not isinstance(response, dict):
+                continue
+
+            task = response["task"]
+
             try:
-                chain_name = task.get("chain")
+                chain_name = task["name"][4:]
```

Comment:
This doesn't work yet as the API doesn't return the name of the provider that is used.

Reply:
This has to be implemented in https://github.com/nextcloud/server/pull/45391